src/relayclient/KnownRelaysManager.ts

```diff
@@ -140,7 +140,9 @@ export default class KnownRelaysManager implements IKnownRelaysManager {
 
   async getRelaysSortedForTransaction (gsnTransactionDetails: GsnTransactionDetails): Promise<RelayInfoUrl[][]> {
     const sortedRelays: RelayInfoUrl[][] = []
-    for (let i = 0; i < this.knownRelays.length; i++) {
+    // preferred relays are kept as-is, unsorted (we don't have any info about them anyway to sort)
```

Comment:
Why does it matter? No data to sort on -> no sorting, are you optimising CPU cycles here? If so, please don't.

Reply:
this IS the bug that was in preferred relay. the root cause is that the sorting code chokes when passed a no-info relayinfo... 
instead, we just need to leave the preferred ones as-is, so that's what I do.

Reply:
Ok, so maybe the sorting code should not choke on no-info, but just leave the items unsorted? 
Why do we even put relays in a two-dimensional array if we treat them as separate variables?

Reply:
we don't. after sorting (which relies on the struct info) we no longer treat them separately. That's why keeping the first line out of sorting solves the bug (try to checkout, remove this fix and see the new test fails..)